trieste/models/gpflow/inducing_point_selectors.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module is the home of  Trieste's functionality for choosing the inducing points
+of sparse variational Gaussian processes.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Generic
+
+import tensorflow as tf
+from scipy.cluster.vq import kmeans
+
+from ...data import Dataset
+from ...space import Box, SearchSpace
+from ...types import TensorType
+from ..interfaces import ProbabilisticModelType, SupportsGetInducingVariables
+
+
+class InducingPointSelector(ABC, Generic[ProbabilisticModelType]):
+    """
+    This class provides functionality to update the inducing points of an inducing point-based model
+    as the Bayesian optimization progresses.
+
+    The only constraint on subclasses of :class:`InducingPointSelector` is that they preserve
+    the shape of the inducing points so not to trigger expensive retracing.
+
+    It can often be beneficial to change the inducing points during optimization, for example
+    to allow the model to focus its limited modelling resources into promising areas of the space.
+    See :cite:`vakili2021scalable` for demonstrations of some of
+    our :class:`InducingPointSelectors`.
+    """
+
+    def __init__(self, search_space: SearchSpace, recalc_every_model_update: bool = True):
+        """
+        :param search_space: The global search space over which the optimization is defined.
+        :param recalc_every_model_update: If True then update the inducing points for each
+            model update, otherwise just update on the first call.
+        """
+
+        self._search_space = search_space
+        self._recalc_every_model_update = recalc_every_model_update
+        self._initialized = False
+
+    def update(
```

Comment:
Not sure update is the best name (it suggest internal state, not returning a value). Why not just

```suggestion
    def calculate_inducing_points(
```